### PR TITLE
flow 0.23

### PR DESCRIPTION
**Pull request by iamdustan**
Seeing @kassens PRs to this repo was always my notification that a new flow release was available. When first upgrading from flow@0.21 -> flow@0.23 there were ~165 errors (exact number I forget, but it was in that range).

Most of it was that flow now requires the `State` shape to be declared on components that use it and other static properties like `node => this.node = node`. After getting through all those rather mundane changes there were a few other changes that required a bit more work.

Specifically the following warning I couldn’t reconcile with only flow definition changes, so I explicitly converted it to a `string` in the `BlurInput.setState` call.

```sh
plugins/ReactNativeStyle/StyleEdit.js:63
 63:           <BlurInput
               ^ React element `BlurInput`
 65:             onChange={newAttr => this.setState({newAttr})}
                                                     ^^^^^^^ number. This type is incompatible with
 26:   newAttr: string,
                ^^^^^^ string
```

Additionally, the `Panel` was being used by the different shells with a `{...config}` object that didn’t have many of the required props. This was solved by marking more fields as optional and putting in the appropriate checks for it.

Lastly `ReactElement` -> `React$Element` in the flowdef. https://github.com/facebook/flow/blob/master/lib/react.js#L106-L116 Simply renaming it seemed to suffice, but there may be details I’m missing.

### CODE REVIEW

File: frontend/DataView/DataView.js [hunk 1]
```diff
@@ -19,18 +19,20 @@ var assign = require('object-assign');
 var consts = require('../../agent/consts');
 var previewComplex = require('./previewComplex');
 
+type DataViewProps = {
+  data: Object,
+  path: Array<string>,
+  inspect: (path: Array<string>, cb: () => void) => void,
+  showMenu: boolean | (e: DOMEvent, val: any, path: Array<string>, name: string) => void,
```
Comment (jaredly) [inline frontend/DataView/DataView.js]: given that showMenu is used twice, maybe make a separate declaration for it?

Comment (iamdustan) [inline frontend/DataView/DataView.js]: done

**Comment by jaredly** [comment]
hmm can you fix travis to install the right version of flow?

**Comment by iamdustan** [comment]
any idea why it’s not? that’s kinda throwing me. I updated it in the package.json and I don’t see any other reason for 0.21 to be coming in instead of 0.23...

**Comment by iamdustan** [comment]
ah. shrinkwrap. Updating now.

**Comment by iamdustan** [comment]
wow shrinkwrap is a PITA. flow is now passing on travis, but something changed in eslint while trying to regenerate the shrinkwrap and now that fails. :(

**Comment by iamdustan** [comment]
:boom: shrinkwrapped properly. CI is back to green!

**Comment by kassens** [comment]
Sorry for the delay here, we should probably just get rid of the shrinkwrap.

**Comment by iamdustan** [comment]
I suppose I can start the update to flow 0.25 now, too :)

**Comment by kassens** [comment]
@iamdustan btw, I created #374 to delete the shrinkwrap file. Should make this easier :)

**Pull request merged**